cmds/core/uptime/uptime.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2013-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Get the time the machine has been up
+//
+// Synopsis:
+//     uptime
+//
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// This function  returns the string current time in the format Hr:Min:Sec
+func currentTime() string {
+	currTime := strings.Split(time.Now().String(), " ")
+	timeArray := strings.Split(currTime[1], ".")
+	return timeArray[0]
+}
+
+// This function takes in the filepath to a file containing loadavgs,it then extracts and returns the three load averages
+func loadavg(loadavgfilepath string) (load1, load2, load3 string, err error) {
+	loadaverages, err := ioutil.ReadFile(loadavgfilepath)
+	if err != nil {
+		return "-1", "-1", "-1", errors.New("Invalid File Contents")
+	}
+	loadAvgArray := strings.Split(string(loadaverages), " ")
+	return loadAvgArray[0], loadAvgArray[1], loadAvgArray[2], nil
+}
+
+// This function takes in the filepath to a file containing the uptime,it then extracts and returns the uptime in the formart
+// Days , Hours , Minutes ,Seconds
+func uptime(uptimefilepath string) (days, hours, minutes int, err error) {
+
+	uptime, err := ioutil.ReadFile(uptimefilepath)
+	if err != nil {
+		return -1, -1, -1, errors.New("Invalid File")
+	}
+	uptimeString := string(uptime)
+	uptimeArray := strings.Split(uptimeString, " ")
+	uptimeSeconds := strings.Split(uptimeArray[0], ".")
+	uptimeInteger, err := strconv.Atoi(uptimeSeconds[0])
+
+	if err != nil {
+		return -1, -1, -1, errors.New("Invalid File Contents")
+	}
+
+	uptimeDuration := time.Duration(uptimeInteger) * time.Second
+	uptimeDays := int(uptimeDuration.Hours() / 24)
+	uptimeHours := int(uptimeDuration.Hours()) - int(uptimeDuration.Hours()/24)*24
+	uptimeMinutes := int(uptimeDuration.Minutes()) - uptimeDays*1440 - uptimeHours*60
+	return uptimeDays, uptimeHours, uptimeMinutes, nil
```

Comment:
This a good idea but I think returning a formatted string makes it harder to test if the return values of this function are correct .

Reply:
OK, there's a slightly easier way to do this that lets you avoid all this math.
Basically, you have created a Duration. You can create a zero-valued time, add the duration to it,  and from there use the basic time methods.

Looks like this:
	// output from /proc/uptime
	u := "22198.65 88695.40"
	a := strings.Fields(u)

	// convert the uptime part of the strings to a Duration.
	// we have to add the s since the number as provided has
	// no units.
	du, _ := time.ParseDuration(string(a[0]) + "s")
 // create a time, from the duration, relative to the zero  value of time,
        t := time.Time{}.Add(du)
	fmt.Println(t.Hour(), t.Minute(), t.Second())
Hour, Minute, and Second are ints, so that is pretty convenient.

https://play.golang.org/p/uRQOfeKi3WB

Reply:
The functions should return strings.

Reply:
Done